daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -879,6 +882,12 @@ object Main {
             var keepParsing = true
             var exitCode = ExitCode.Success
 
+            if (parseOpts.stream.toOption.getOrElse(false) && (infosetType == InfosetType.EXI)) {
+              exitCode = ExitCode.NotYetImplemented
+              keepParsing = false
+              System.err.println("Streaming mode with EXI infosets is not currently supported")
+            }
```

Comment:
This only errors when parsing. Streaming can also be enabled for unparsing so we should detect that too.

Also, maybe a cleaner way to check the options is using scallops `validateOpt` function, I think something like this:

```scala
validateOpt(infosetType, stream) {
  case (Some(InfosetType.EXI), Some(true)) => Left("Error message here ... ")
  case _ => Right(Unit)
}
```

We use something similar a few places in the scallop configuration.